mapit_labour/management/commands/mapit_labour_import_addressbase_core.py

```diff
@@ -85,9 +93,13 @@ def handle_label(self, label: str, **options):
         self.purge = options["purge"]
         self.batch_size = options["batch_size"]
         self.limit = options["limit"]
+        self.dry_run = options["dry_run"]
 
         with open_compressed_maybe(label, mode="rt", encoding="utf-8-sig") as f:
-            self.handle_rows(DictReader(f))
+            with transaction.atomic():
+                self.handle_rows(DictReader(f))
+                if self.dry_run:
+                    transaction.set_rollback(True)
```

Comment:
Is fine, just noting that this would 'lose' any primary IDs added to the DB as part of this (because they can never be reused), if that was a concern in production (e.g. I'd try to avoid something like this on live mapit, doesn't matter in dev presumably). Alternative would be to skip the bulk_create/update lines if dry-run presumably, I don't know if it's important to test the db lines actually run or if we can assume they do.

Reply:
I generally prefer the rollback method rather than skipping DB queries because a) you get a feel for how fast things will actually go (this is what led me to the big speed up in ) and b) don't get any nasty surprises from DB constraints when running for real.

In this case the PK is the UPRN as opposed to a `serial` so no worries there.